### PR TITLE
Added tests

### PR DESCRIPTION
**Pull request by mhorianin**

### CODE REVIEW

File: src/test/resources/database/remove-saved-category-from-db.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+DELETE FROM categories WHERE name = 'Verse';
```
Comment (okuzan) [inline src/test/resources/database/remove-saved-category-from-db.sql]: i'd suggest forming a habit and always reviewing your PR to fix no empty line at EOF issues
image

File: src/test/java/com/example/bookstoreweb/repository/BookRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Test
+    @DisplayName("Find all books by category id")
+    @Sql(scripts = "classpath:database/add-book-to-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/remove-data-from-all-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void findAllByCategoryId_ValidId_ReturnOneBook() {
```
Comment (okuzan) [inline src/test/java/com/example/bookstoreweb/repository/BookRepositoryTest.java:25]: some of your test methods are public, some are not, better unify this

File: src/test/java/com/example/bookstoreweb/repository/BookRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+class BookRepositoryTest {
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Test
+    @DisplayName("Find all books by category id")
+    @Sql(scripts = "classpath:database/add-book-to-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/remove-data-from-all-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
```
Comment (okuzan) [inline src/test/java/com/example/bookstoreweb/repository/BookRepositoryTest.java]: common to both of these tests, extract

File: src/test/java/com/example/bookstoreweb/controller/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+    @Test
+    @WithMockUser(username = "user", authorities = {"USER"})
+    @DisplayName("Get a list of all categories")
+    @Sql(scripts = "classpath:database/add-category-for-book.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/remove-data-from-all-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void findAllCategories_Success() throws Exception {
+        MvcResult result = mockMvc.perform(get("/api/categories"))
```
Comment (okuzan) [inline src/test/java/com/example/bookstoreweb/controller/CategoryControllerTest.java]: it's a good idea to extract base url to a constant, so you only ever have to change it in one place, if needed

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/bookstoreweb/repository/BookRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Test
+    @DisplayName("Find all books by category id")
+    @Sql(scripts = SCRIPT_FOR_ADD_DATA_IN_DB,
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = SCRIPT_FOR_REMOVE_DATA_IN_DB,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void findAllByCategoryId_ValidId_ReturnOneBook() {
```
Comment (okuzan) [inline src/test/java/com/example/bookstoreweb/repository/BookRepositoryTest.java]: image
sorry for not being clear enough, I meant extract to a class level

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Rommelua — APPROVED** [review]

**Pull request merged**